Tests/Unit/EmogrifierTest.php

```diff
@@ -750,6 +756,43 @@ public function emogrifyFormatsCssDeclarations($cssDeclarationBlock, $expectedSt
     }
 
     /**
+     * Data provider for emogrifyInvalidDeclaration.
+     *
+     * @return array[]
+     */
+    public function invalidDeclarationDataProvider()
+    {
+        return [
+            'one declaration with missing dash in property name' => ['font weight: bold;', 'weight:'],
+            'one declaration with invalid character in property name' => ['-9webkit-text-size-adjust:none;', 'webkit'],
+            'one declaration with missing :' => ['-webkit-text-size-adjust none', 'webkit'],
+            'one declaration with missing value' => ['-webkit-text-size-adjust :', 'webkit'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $cssDeclarationBlock the CSS declaration block (without the curly braces)
+     * @param string $notExpectedStyleAttributeContent the value that should not be withint the style attribute
+     *
+     * @dataProvider invalidDeclarationDataProvider
+     */
+    public function emogrifyInvalidDeclaration($cssDeclarationBlock, $notExpectedStyleAttributeContent)
+    {
+        $html = $this->html5DocumentType . '<html></html>';
+        $css = 'html {' . $cssDeclarationBlock . '}';
+
+        $this->subject->setHtml($html);
+        $this->subject->setCss($css);
+
+        self::assertNotContains(
+            $notExpectedStyleAttributeContent,
+            $this->subject->emogrify()
```

Comment:
Okay, the style node then should not be there in the first place. Let's tackle this separately from this PR. I've entered an issue for this: #259